xds/src/main/java/io/grpc/xds/internal/security/trust/XdsTrustManagerFactory.java

```diff
@@ -100,6 +122,23 @@ private static X509Certificate[] getTrustedCaFromCertContext(
   @VisibleForTesting
   static XdsX509TrustManager createX509TrustManager(
       X509Certificate[] certs, CertificateValidationContext certContext) throws CertStoreException {
+    return new XdsX509TrustManager(certContext, createTrustManager(certs));
+  }
+
+  @VisibleForTesting
+  static XdsX509TrustManager createX509TrustManager(Map<String, List<X509Certificate>> spiffeRoots,
+      CertificateValidationContext certContext) throws CertStoreException {
+    checkNotNull(spiffeRoots, "spiffeRoots");
+    Map<String, X509ExtendedTrustManager> delegates = new HashMap<>();
+    for (Map.Entry<String, List<X509Certificate>> entry:spiffeRoots.entrySet()) {
+      delegates.put(entry.getKey(), createTrustManager(
+          entry.getValue().toArray(new X509Certificate[0])));
```

Comment:
Is this assuming that we will only have 1 value per key? Apologies in advance if I'm completely misreading this.

Reply:
Thanks for clarifying!